File: 1894-merge-strings-alternately/merge-strings-alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    longerWord = word1.length > word2.length ? word1 : word2
    mergedStr = ""

    for(let i = 0; i < longerWord.length; i++) {
        if (word1[i]) {
            mergedStr += word1[i]
        }
        if (word2[i]) {
            mergedStr += word2[i]
        }
    }

    return mergedStr
};

File: 1894-merge-strings-alternately/merge-strings-alternately.rb
# @param {String} word1
# @param {String} word2
# @return {String}
def merge_alternately(word1, word2)
    # Inputs: word1 and word2, strings
    # output: 1 merged string
    result = ""
    longest_word_length = [word1.size, word2.size].max
    (0..longest_word_length).each do |i|
        result << word1[i] if word1[i]
        result << word2[i] if word2[i]
    end
    result
end